test/setup.ts

```diff
@@ -0,0 +1,50 @@
+import 'core-js/proposals/reflect-metadata'
```

Comment:
So I would probably recommend now taking this approach for unit tests, I think it makes you have to setup a sort of a parallel universe to all the app things eventually, which isn't ideal for small tests. One of the nice things about our injection setup is that we have regular constructor params to use, so you could actually just call them directly.

I'd probably pull these fakes into another file somewhere (maybe a `test/fakes/` folder or something?) and not setup the injection at all, instead constructing them manually.

The goal here is to fake as few things as possible. I definitely think you'll need to fake out `NydusServer` since it has so many external deps, but I think `UserSocketsManager`/`ClientSocketsManager` are probably fine with no faking, provided you integrate a fake session lookup.